overlord/boot/firstboot.go

```diff
@@ -122,14 +123,20 @@ func populateStateFromInstalled() error {
 	return ovld.Stop()
 }
 
+// replaced in the tests
+var firstbootEnableFirstEther = firstboot.EnableFirstEther
+
 // FirstBoot will do some initial boot setup and then sync the
 // state
 func FirstBoot() error {
 	if firstboot.HasRun() {
 		return ErrNotFirstBoot
 	}
-	if err := firstboot.EnableFirstEther(); err != nil {
-		logger.Noticef("Failed to bring up ethernet: %s", err)
+
+	if !release.OnClassic {
+		if err := firstbootEnableFirstEther(); err != nil {
+			logger.Noticef("Failed to bring up ethernet: %s", err)
+		}
 	}
 
 	// snappy will be in a very unhappy state if this happens,
```

Comment:
May I ask an irrelative question to this pull request?
Is there any reason that the system need to enable first ether on firstboot?

I trace the code and found it is not fatal error if enable failed. But system with ethernet not connecting to cable will stuck in dhclient for several minutes at first boot up. (I have leveraged series 16 ubuntu-core image)